core/templates/dev/head/player/PlayerServices.js

```diff
@@ -197,31 +196,17 @@ oppia.factory('oppiaPlayerService', [
 
     _stopwatch.resetStopwatch();
 
-    var newStateData = _exploration.states[newStateName];
     $rootScope.$broadcast('playerStateChange');
     successCallback(
       newStateName, refreshInteraction, newFeedbackHtml, newQuestionHtml,
       newParams);
   };
 
-  var _registerMaybeLeaveEvent = function(stateName) {
-    var maybeLeaveExplorationUrl = (
-      '/explorehandler/exploration_maybe_leave_event/' + _explorationId);
-    $http.post(maybeLeaveExplorationUrl, {
-      client_time_spent_in_secs: _stopwatch.getTimeInSecs(),
-      params: learnerParamsService.getAllParams(),
-      session_id: sessionId,
-      state_name: stateName,
-      version: version
-    });
-  };
-
-  // This should only be called when _exploration is non-null.
+  // This should only be called when 'exploration' is non-null.
   var _loadInitialState = function(successCallback) {
-    var initStateName = _exploration.init_state_name;
     var initStateData = stateTransitionService.getInitStateData(
-      _exploration.param_specs, _exploration.param_changes,
-      _exploration.states[initStateName]);
+      exploration.paramSpecs, exploration.paramChanges,
```

Comment:
I lean against this (moving methods from stateTransitionService into ExplorationObjectFactory). I think we should keep the domain objects as thin as possible. I originally was just expecting there to be trivial getters and nothing more. I think the amount of 'logic' the current one does is as much as it should. The factory basically represents the exploration data structure used by the entire frontend. I think the functions in stateTransitionService are too specific to displaying explorations to belong in such a generic object factory.